File: LeetCode/java/2259_Remove_Digit_From_Number_to_Maximize_Result.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

class Solution {
    public String removeDigit(String number, char digit) {
        String answer = "";
        List<Integer> idx = new ArrayList<>();
        List<String> list = new ArrayList<>();

        for (int i = 0; i < number.length(); i++) {
            if (number.charAt(i) == digit) {
                idx.add(i);
            }
        }

        for (int i : idx) {
            StringBuilder temp = new StringBuilder(number);

            temp.deleteCharAt(i);
            list.add(temp.toString());
        }

        Collections.sort(list);

        answer = list.get(list.size() - 1);

        return answer;
    }
}

File: LeetCode/cpp/2259_Remove_Digit_From_Number_to_Maximize_Result.cpp
class Solution {
public:
	string removeDigit(string number, char digit) {
		string answer = "";
		vector<int> idx;
		vector<string> v;

		for (int i = 0; i < number.length(); i++) {
			if (number[i] == digit) {
				idx.push_back(i);
			}
		}

		for (int i : idx) {
			string temp = number;

			temp.erase(temp.begin() + i);
			v.push_back(temp);
		}

		sort(v.begin(), v.end());

		answer = v[v.size() - 1];

		return answer;
	}
};